plugins/content/courseasset/index.js

```diff
@@ -42,38 +42,67 @@ CourseAssetContent.prototype.getModelName = function () {
 function initialize() {
   // Content Hook for updatedAt and updatedBy:
   ['course', 'contentobject', 'article', 'block', 'component'].forEach(function (contentType) {
-    app.contentmanager.addContentHook('destroy', contentType, { when: 'pre' }, function (contentType, data, next) { 
-
+    app.contentmanager.addContentHook('destroy', contentType, { when: 'pre' }, function (contentType, data, next) {
       database.getDatabase(function (err, db) {
         db.retrieve(contentType, data[0], function(err, items) {
           if (err) {
             logger.log('err', err);
             return next(err);
           }
-
           if (items && items.length == 1) {
             var itemForDeletion = items[0].toObject();
             var criteria = {};
-            
+
             async.series([
               function(callback){
+                if (contentType !== 'contentobject') {
+                  return callback(null, 'Not processing a content object');
+                }
+                db.retrieve('article', { _courseId: itemForDeletion._courseId, _parentId: itemForDeletion._id }, function(err, articles) {
+                  if (err) {
+                    return callback(err, 'Unable to retrieve child articles of contentobject ' + itemForDeletion._id);
+                  }
+                  var parentIds = [];
+                  if (articles && articles.length !== 0) {
+                    criteria._courseId = itemForDeletion._courseId;
+                    parentIds = _.pluck(articles, '_id');
+                  }
+                  async.each(articles, function(article, cb) {
+                    db.retrieve('block', {_courseId: article._courseId, _parentId: article._id}, function(err, blocks) {
+                      if (err) {
+                        return callback(err, 'Unable to retrieve child blocks of article ' + article._id);
+                      }
+                      if (blocks && blocks.length !== 0) {
+                        parentIds = parentIds.concat(_.pluck(blocks, '_id'));
+                      }
+                      if(parentIds.length > 0) {
+                        criteria.$or = [
+                          { _contentTypeParentId: { $in: parentIds } },
+                          { _contentTypeId: itemForDeletion._id }
+                        ];
+                      } else {
+                        criteria._contentTypeId = itemForDeletion._id;
+                      }
+                      callback(null, 'Child content objects added to criteria object');
```

Comment:
Getting header errors with multiple articles or blocks possible due to this.
`Error: Can't set headers after they are sent.`